### PR TITLE
Better message text for ResponseException

### PR DESCRIPTION
**Pull request by synhershko**
This avoids messages with malformed URLs, like `org.elasticsearch.client.ResponseException: PUT http://127.0.0.1:9502customer: HTTP/1.1 400 Bad Request`

### CODE REVIEW

**Comment by jasontedor** [comment]
Thanks for the PR. Would mind providing a reproduction that shows when this can happen?

**Comment by synhershko** [comment]
I have it happening with the following code:

```
Response response = restClient.performRequest("PUT", "customer", newHashMap(), new NStringEntity("{\"invalid\":0}", ContentType.APPLICATION_JSON));
```

Because "customer" is without a leading `/`, and the hosts in RestClient I assume will never hold trailing `/`, a message like in the PR description happens.

If you want this within an actual test it might take me a few days to provide

**Comment by jasontedor** [comment]
Thanks, no need for a test for a situation like this.

**Comment by jasontedor** [comment]
So I don't think this is how we should fix this, I would prefer that we not "quietly" imply a "/" where none was given (in this case, neither the hostname nor the URI). Instead, I would suggest it be changed to something like:

```
method [PUT], host [http://127.0.0.1:9502], URI [customer], status [HTTP/1.1 400 Bad Request]
```

**Comment by synhershko** [comment]
Makes sense, will do!

**Comment by synhershko** [comment]
@javanna @jasontedor ready for review - also moved to using StringBuilder, which seems the common practice in the code base

File: client/rest/src/main/java/org/elasticsearch/client/ResponseException.java [hunk 1]
```diff
@@ -39,18 +39,21 @@ public ResponseException(Response response) throws IOException {
     }
 
     private static String buildMessage(Response response) throws IOException {
-        String message = response.getRequestLine().getMethod() + " " + response.getHost() + response.getRequestLine().getUri()
-                + ": " + response.getStatusLine().toString();
+        final StringBuilder sb = new StringBuilder("method [");
+        sb.append(response.getRequestLine().getMethod());
+        sb.append("], host [").append(response.getHost());
+        sb.append("], URI [").append(response.getRequestLine().getUri());
+        sb.append("], status line [").append(response.getStatusLine().toString()).append(']');
```
Comment (jasontedor) [inline client/rest/src/main/java/org/elasticsearch/client/ResponseException.java]: I appreciate the use of `StringBuilder`, but I think we can do something simpler and clearer here:

```java
String.format(
    Locale.ROOT,
    "method [%s], host [%s], URI [%s], status line [%s]\n%s",
    . 
    .
    .);
```

I find this so much easier to see what the end result is going to be.

**Review by jasontedor — CHANGES_REQUESTED** [review]
Thanks for the iteration @synhershko, I left another suggestion.

**Comment by synhershko** [comment]
Bah that's the first thing I did.. will update it back again now

**Comment by synhershko** [comment]
done

**Comment by jasontedor** [comment]
Looks good, I'll start a CI run. Thanks.

**Comment by jasontedor** [comment]
test this please

**Comment by jasontedor** [comment]
@synhershko It looks like there is already a test for this and it's failing with the change. Would you mind updating the test?

**Comment by synhershko** [comment]
@jasontedor done

**Comment by jasontedor** [comment]
test this please

**Comment by jasontedor** [comment]
test this please

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by jasontedor** [comment]
Thanks @synhershko!

**Comment by synhershko** [comment]
Awesome! @jasontedor any chance you can backport this to 6.x and 5.x, where I first encountered this issue?

**Comment by jasontedor** [comment]
@synhershko I did backport this to 6.x and 6.0 already. I need to think about backporting to 5.6.

**Comment by jasontedor** [comment]
I think this should remain as-is in 5.6.

**Pull request merged**